Packages/com.unity.inputsystem/InputSystem/Devices/Remote/InputRemoting.cs

```diff
@@ -499,29 +517,48 @@ public struct Data
             {
                 public string name;
                 public string layout;
+                public string layoutJson;
                 public int deviceId;
                 public InputDeviceDescription description;
             }
 
-            public static Message Create(InputDevice device)
+            public static Message Create(InputRemoting sender, InputDevice device)
             {
                 Debug.Assert(!device.remote, "Device being sent to remotes should be a local device, not a remote one");
 
+                // Try to load the layout. Ignore the layout if it couldn't
+                // be loaded.
+                InputControlLayout layout = null;
+                try
+                {
+                    layout = sender.m_LocalManager.TryLoadControlLayout(new InternedString(device.layout));
+                    if (layout == null)
+                    {
+                        Debug.Log(string.Format(
+                            "Could not find layout '{0}' associated with device '{1}' meant to be sent through remote connection; this should not happen",
+                            device.layout, device.name));
+                    }
+                }
+                catch (Exception exception)
+                {
+                    Debug.Log(string.Format(
+                        "Could not load layout '{0}' associated with device '{1}'; not sending to remote listeners (exception: {2})",
+                        device.layout, device.name, exception));
+                }
+
                 var data = new Data
                 {
                     name = device.name,
                     layout = device.layout,
+                    layoutJson = layout?.ToJson(),
                     deviceId = device.deviceId,
                     description = device.description
                 };
 
-                var json = JsonUtility.ToJson(data);
-                var bytes = Encoding.UTF8.GetBytes(json);
-
                 return new Message
                 {
                     type = MessageType.NewDevice,
-                    data = bytes
+                    data = SerializeData(data)
```

Comment:
guess we're going to have to get this tested on all platforms? as this could impact mobile et al?